### PR TITLE
Fix many docfx warnings

### LINKED ISSUE #1067: docfx doesn't recognize Microsoft.* namespaces in xrefs
Build produces "invalid cross reference" warnings for the following references:
| File | Ref type |
| :--- | :--- |
middleware.md | Microsoft.AspNetCore.Builder.ReverseProxyIEndpointRouteBuilderExtensions.MapReverseProxy
dests-health-checks.md | Microsoft.AspNetCore.Builder.ProxyMiddlewareAppBuilderExtensions
transforms.md | Microsoft.Extensions.DependencyInjection.ReverseProxyServiceCollectionExtensions

### PR DESCRIPTION
**Pull request by alnikola**
Fixes most of the docfx warnings. It leaves only 5 active warnings which complain about missing metadata references in `src/OperatorFramework/examples/BasicOperator/BasicOperator.csproj`. They can be ignored for now.

Contributes to #1067

### CODE REVIEW

File: src/OperatorFramework/src/Core/GroupKindNamespacedName.cs [hunk 1]
```diff
@@ -87,21 +87,18 @@ public static GroupKindNamespacedName From(IKubernetesObject<V1ObjectMeta> resou
                 NamespacedName.From(resource));
         }
 
-        /// <inheritdoc/>
```
Comment (Tratcher) [inline src/OperatorFramework/src/Core/GroupKindNamespacedName.cs:90]: What's wrong with inheritdoc?

**Review by Tratcher — APPROVED** [review]
Should that be "contributes to #"? If we're not going to fix all of the warnings yet then the issue should stay open, even if moved to the backlog.

File: src/OperatorFramework/src/Core/GroupKindNamespacedName.cs (hunk 1 above)
Comment (alnikola) [reply src/OperatorFramework/src/Core/GroupKindNamespacedName.cs:90]: It doesn't find all of them. I removed the ones docfx complained about.

**Comment by alnikola** [comment]
>If we're not going to fix all of the warnings yet then the issue should stay open, even if moved to the backlog.

@Tratcher Are you sure we are going to fix them for 1.0? It's 1 project inside Kubernetes integration package.

**Comment by Tratcher** [comment]
We don't have to fix them for 1.0, but the tracking issue should stay open and indicate what work remains. We can re-triage the issue to decide when to do it.

**Comment by alnikola** [comment]
> We don't have to fix them for 1.0, but the tracking issue should stay open and indicate what work remains.
Got it. I changed it to Contributes to.

**Pull request merged**